src/daemon/daemon.cpp

```diff
@@ -2131,17 +2112,66 @@ mp::Daemon::async_wait_for_ssh_and_start_mounts(grpc::ServerWriter<Reply>* serve
                 }
                 catch (const mp::SSHFSMissingError&)
                 {
-                    fmt::format_to(errors, "Error enabling mount support in '{}'", name);
+                    fmt::format_to(errors, "Error enabling mount support in '{}'\n", name);
                     break;
                 }
             }
             catch (const std::exception& e)
             {
-                fmt::format_to(errors, "Removing \"{}\": {}", target_path, e.what());
+                fmt::format_to(errors, "Removing \"{}\": {}\n", target_path, e.what());
                 invalid_mounts.push_back(target_path);
             }
         }
     }
+    catch (const std::exception& e)
+    {
+        fmt::format_to(errors, e.what());
+    }
+
+    return fmt::to_string(errors);
+}
+
+template <typename Reply>
+mp::Daemon::AsyncOperationStatus mp::Daemon::async_wait_for_ready_all(grpc::ServerWriter<Reply>* server,
+                                                                      const std::vector<std::string>& vms,
+                                                                      std::promise<grpc::Status>* status_promise)
+{
+    QFutureSynchronizer<std::string> start_synchronizer;
+    for (const auto& name : vms)
+    {
+        std::lock_guard<decltype(start_mutex)> lock{start_mutex};
```

Comment:
Instead of locking and unlocking for each iteration of the loop, why not lock outside the for loop?

Reply:
Sure, I can do that.  I was just trying to take advantage of the automatic unlock in the scope of the `for` loop, but sure I can just add a new scope encapsulating the `for` loop and only do one lock for the whole thing.

Reply:
I moved this into its own scope and moved the lock outside of the `for` loop.